docs/semantics.md

```diff
@@ -163,7 +163,27 @@ Additional file cache [behavior](https://cloud.google.com/storage/docs/gcsfuse-c
 
    - If a Cloud Storage FUSE client modifies a cached file or its metadata, then the file is immediately invalidated and consistency is ensured in the following read by the same client. However, if different clients access the same file or its metadata, and its entries are cached, then the cached version of the file or metadata is read and not the updated version until the file is invalidated by that specific client's TTL setting.     
 
-**Note**: 
+**Kernel List Cache**
+
+The Cloud Storage FUSE kernel-list-cache is used to cache the directory listing (output of `ls`). It significantly improves the workload which involves repeated listing. This is recommended to use it with read-only mounts specifically for Serving and Training workload.
+
+By default, the list cache is disabled. It can be enabled by configuring the **--kernel-list-cache-ttl-secs** cli flag or **file-system:kernel-list-cache-ttl-secs** config flag where:
+*   a value of 0 means disabled. This is the default value.
+*   A positive value represents the ttl (in seconds) to keep the directory list response in the kernel page-cache.
+*   -1 to bypass entry expiration and always return the list response from the cache if available.
+
+**Important Points**
+*   The kernel-list-cache is kept within the page cache. Consequently, this functionality depends upon the availability of page cache memory on the system. This contrasts with the stat and type caches, which are retained in user memory as part of Cloud Storage Fuse daemon.
+*   The kernel's list cache is maintained on a per-directory level, resulting in either all list entries being retained in the page cache or none at all.
+*   The creation, renaming, or deletion of new files or folders causes the eviction of the page-cache of their immediate parent directory, but not of all ancestral directories.
+
+**Consistency**
+*   Kernel List cache ensures consistency within the mount. That means, creation, deletion or rename of files/folder within a directory evicts the kernel list cache of the directory.
+*   Externally added objects are only visible after the kernel-list-cache-ttl-secs timer expires, even if they are touched (stat) via the same mount point. Since stat doesn’t evict the kernel-list-cache so there might be some list-stat inconsistency.
+*   Kernel-list-cache-ttl doesn't work with empty directories. Once an empty directory list response is cached, it requires a new file to be created from the same mount point to invalidate the cache.
```

Comment:
this line doesn't make sense to me. 

I guess we should add that "In case another external client adds a new file to the empty directory, the client1 is still not going to get the new file even after the TTL is expired"